tests/testthat/test-verb-pivot-longer.R

```diff
@@ -0,0 +1,258 @@
+test_that("can pivot all cols to long", {
```

Comment:
Again, I'd prefer to start with some simple tests, mostly focussing on the SQL generation, and with only a few checking the actual execution. I think we'll also have to tackle the tests in pieces so that I can progressively build up an understanding of what's going on.